src/routes/auth.ts

```diff
@@ -61,46 +91,63 @@ export default (database: DbConnection) => {
   });
 
   router.post("/getkey", async (req, res) => {
-    const { username, password }: { username: string, password: string } = req.body;
+    const {
+      username,
+      password,
+    }: { username: string; password: string } = req.body;
     if (!username || !password) return res.sendStatus(400);
 
     if (username.length > 20) return res.sendStatus(400);
 
-    let dbUser = await database.repositories.User.findOne({ where: { name: username } });
+    let dbUser = await database.repositories.User.findOne({
+      where: { name: username },
+    });
     if (!dbUser) return res.sendStatus(404);
 
     let passwordMatch = await verify(password, dbUser.password);
     if (!passwordMatch) return res.sendStatus(401);
 
+    let token = dbUser.apiKey;
+    if (token.startsWith("$")) token = await eToken.decrypt(token, password);
```

Comment:
Was wondering if we could validate more specifically instead of comparing with a `$` start.
```
$algorithm-identifier$iv-and-salt:auth-tag:encrypted-token:hashed-token$
```

Reply:
Well, tokens in the DB can take one of two forms: Unencrypted, in which case they can not possibly include a $, and encrypted ones, where $ will always be at the start. This is the easiest way to safely check if a token is encrypted

Reply:
Got it :+1: